libcudacxx/test/libcudacxx/cuda/iterators/strided_iterator/subscript.pass.cpp

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// constexpr reference operator[](difference_type n) noexcept;
+// constexpr const_reference operator[](difference_type n) const noexcept;
+
+#include <cuda/iterator>
+#include <cuda/std/cassert>
+
+#include "test_macros.h"
+#include "types.h"
+
+template <class Stride>
+__host__ __device__ constexpr void test(Stride stride)
+{
+  int buffer[] = {1, 2, 3, 4, 5, 6, 7, 8};
+  {
+    cuda::strided_iterator iter{buffer, stride};
+    assert(iter[4] == 5);
+    assert(cuda::std::addressof(iter[4]) == buffer + 4);
```

Comment:
Critical:
```suggestion
  int buffer[] = {1, 2, 3, 4, 5, 6, 7, 8};
  {
    cuda::strided_iterator iter{buffer, stride};
    assert(iter[2] == 5);
    assert(cuda::std::addressof(iter[2]) == buffer + 4);
```
Probably more problems in this test.

Reply:
Yeah that was a remnant from the bug in subscript